### PR TITLE
[ONNX] Fix circular padding to support dynamic axes

### PR DESCRIPTION
**Pull request by ilyasher**
This commit fixes a bug where the ONNX exporter for circular padding queried the input tensor shape in order to get the correct 'end' index for a slice node. This doesn't work when the axis in question is has dynamic size. The commit fixes this by setting the 'end' index to INT_MAX, which is the recommended way of slicing to the end of a dimension with unknown size per ONNX spec.

See https://onnx.ai/onnx/operators/onnx__Slice.html

Also adds a regression test.

### CODE REVIEW

**Review by BowenBao — APPROVED** [review]
LGTM, thanks for your contribution! You need to sign EasyCLA for me to merge it.

**Comment by ilyasher** [comment]
> LGTM, thanks for your contribution! You need to sign EasyCLA for me to merge it.

Thanks, I'm trying to get that sorted out right now.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -7557,6 +7557,21 @@ def forward(self, x):
...
+            def forward(self, x):
+                out = torch.nn.functional.pad(x, (2, 1, 2, 1), mode="circular")
+                return out
+
+        x = torch.randn(4, 3, 5, 6)
+        self.run_test(
+            PadModel(),
+            x,
+            input_names=["input_1"],
+            dynamic_axes={"input_1": [0, 1, 2, 3]}
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:7572]:
```suggestion
            dynamic_axes={"input_1": [0, 1, 2, 3]},
```

Caught by lintrunner. You could also check for this locally. See https://github.com/pytorch/pytorch/wiki/lintrunner.

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:7572]: Other than this it's only EasyCLA and we are good to go.

Comment (ilyasher) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: I am waiting for the CLA manager at my company to do the paperwork, hopefully it won't take longer than a couple of days.

**Comment by BowenBao** [comment]
Somehow it still shows 
> ❌ The commit (https://github.com/pytorch/pytorch/commit/d72682bbfd74323ce00c5458097ce4459686b50c). This user is missing the User's ID, preventing the EasyCLA check.

I wonder if it's because the other commit has a different author email address.

**Comment by ilyasher** [comment]
> I wonder if it's because the other commit has a different author email address.

Good thinking, I will complete the CLA with my other email.

**Comment by ilyasher** [comment]
The commit shows my email as
Author: Ilya Sherstyuk <46343317+ilyasher@users.noreply.github.com>

I will fix the commit with the correct email and force-push

**Comment by ilyasher** [comment]
@BowenBao I finally got the CLA approved, what else do I need to do to merge this PR? Thanks for all your help.
